sklearn/model_selection/tests/test_search.py

```diff
@@ -1175,12 +1175,30 @@ def test_grid_search_cv_splits_consistency():
                        cv=KFold(n_splits=n_splits))
     gs2.fit(X, y)
 
+    # Give generator as a cv parameter
```

Comment:
To be sure, we've not got anything that ensures this will remain a generator. Either use a generator expression or test for its type.